utils/errors/error.go

```diff
@@ -2,6 +2,7 @@ package errors
 
 import "fmt"
 
+// ConvertPanicToError returns an error based on different types
```

Comment:
Could you improve the description here a bit? The func tends to catch panics in recovery method and change it to an error object